### PR TITLE
Revise documentation to use library agnostic reactive APIs.

### LINKED ISSUE #792: Revise documentation to use library agnostic reactive APIs
### Issue description

Examples and text should use library agnostic APIs as much as possible, and prefer Project Reactor where necessary.

### PR DESCRIPTION
**Pull request by wetted**
closes #792

### CODE REVIEW

**Comment by wetted** [comment]
@sdelamo @jeremyg484 

I have some gaps in my Kafka/Reactive knowledge, so please pay attention to the code examples. I'm not convinced I have them right, and I suspect the method signatures in `kafkaClientMethods.adoc` examples might not work using a `Publisher` argument (`Flux` or `Mono` is likely more correct).

File: kafka/src/test/groovy/io/micronaut/configuration/kafka/docs/consumer/batch/BookListener.java [hunk 1]
```diff
@@ -28,7 +29,7 @@ public void receiveList(List<Book> books) { // <2>
 
     // tag::reactive[]
     @Topic("all-the-books")
-    public Flux<Book> receiveFlux(Flux<Book> books) {
+    public Publisher<Book> receiveFlux(Flux<Book> books) {
```
Comment (jeremyg484) [inline kafka/src/test/groovy/io/micronaut/configuration/kafka/docs/consumer/batch/BookListener.java:32]: The examples in this style with mixed usage of `Flux/Mono` and `Publisher` don't quite sit right with me. If a class is already importing Flux or Mono so as to use their more helpful features, what is the benefit of dropping down and using `Publisher` for the return type? I think I'd prefer to see this continue to return `Flux`.

File: kafka/src/test/groovy/io/micronaut/configuration/kafka/docs/consumer/reactive/ProductListener.java [hunk 2]
```diff
@@ -4,15 +4,18 @@
...
 @KafkaListener
 public class ProductListener {
 
     // tag::method[]
     @Topic("reactive-products")
-    public Mono<Product> receive(@KafkaKey String brand,  // <1>
-                                 Mono<Product> productFlowable) { // <2>
+    @SingleResult
+    public Publisher<Product> receive(@KafkaKey String brand,  // <1>
+                                      Mono<Product> productFlowable) { // <2>
```
Comment (jeremyg484) [inline kafka/src/test/groovy/io/micronaut/configuration/kafka/docs/consumer/reactive/ProductListener.java:18]: Same concern as above re: mixed usage of `Publisher` and `Flux/Mono`.

File: kafka/src/test/groovy/io/micronaut/configuration/kafka/docs/consumer/sendto/ProductListener.java [hunk 3]
```diff
@@ -26,8 +28,9 @@ public int receive(@KafkaKey String brand,
     // tag::reactive[]
     @Topic("awesome-products") // <1>
     @SendTo("product-quantities") // <2>
-    public Mono<Integer> receiveProduct(@KafkaKey String brand,
-                                        Mono<Product> productSingle) {
+    @SingleResult
+    public Publisher<Integer> receiveProduct(@KafkaKey String brand,
+                                             Mono<Product> productSingle) {
```
Comment (jeremyg484) [inline kafka/src/test/groovy/io/micronaut/configuration/kafka/docs/consumer/sendto/ProductListener.java:33]: Same concern as above re: mixed usage of `Publisher` and `Flux/Mono`.

File: kafka/src/test/groovy/io/micronaut/configuration/kafka/docs/consumer/batch/BookListener.java (hunk 1 above)
Comment (wetted) [reply kafka/src/test/groovy/io/micronaut/configuration/kafka/docs/consumer/batch/BookListener.java:32]: OK. Thanks you. I don't disagree, but will see what @Sergio wants to do. He would like to minimize use of library-specific PAIs but I don't always see how to get around it for examples.

Comment (jeremyg484) [reply kafka/src/test/groovy/io/micronaut/configuration/kafka/docs/consumer/batch/BookListener.java:32]: I think you tagged the wrong Sergio. :)

**Comment by sdelamo** [comment]
closed via https://github.com/micronaut-projects/micronaut-kafka/pull/850

**Pull request closed without merging**